csrc/scheduler/transpose.cpp

```diff
@@ -1236,6 +1257,7 @@ void scheduleTranspose(Fusion* fusion, TransposeParams params) {
 
   int pos = (int)reference2->nDims() - 2;
   // [..., tile1, tile2]
+  cleanInnerNDLeafDomain(reference2, 2);
```

Comment:
Might this interfere with `inlineMost`? If you have a setting like this:
```
Inputs:
    T0[ i0 i1 ]
    T1[ i2 ]
Outputs:
    T4[ i0 ]
T2[ i0 i1 ] = - T0[ i0 i1 ]
T3[ i0 r1 ] = sum(T2[ i0 i1 ])
T4[ i0 ] = T3[ i0 r1 ] * T1[ i2 ]
```
I'd expect this to generate these loops:
```
FOR i0
  // Allocate T3
  FOR i1
    T2[ i0 i1 ] = - T0[ i0 i1 ]
    T3[ i0 r1 ] += T2[ i0 i1 ]
  ENDFOR i1
  T4[ i0 ] = T3[ i0 r1 ] * T1[ i2 ]
ENDFOR i0
```
When you reorder it as `T2[ r3 i1 i2 ]` will it break the inlining into multiple outer loops like below?
```
// Allocate T3
FOR i0
  // Allocate T3
  FOR i1
    T2[ i0 i1 ] = - T0[ i0 i1 ]
  ENDFOR i1
ENDFOR i0
FOR i0
  FOR i1
    T3[ i0 r1 ] += T2[ i0 i1 ]
  ENDFOR i1
ENDFOR i0
FOR i0
  T4[ i0 ] = T3[ i0 r1 ] * T1[ i2 ]
ENDFOR i0
```

Reply:
Sorry I know this wasn't a transpose example. Is there something about the fusions this scheduler accepts that means we can ignore this?

Reply:
:stuck_out_tongue_closed_eyes: I don't know enough about how indexing works to answer this one.

```
void cleanInnerNDLeafDomain(TensorView* tv, int n) {
  if (!tv->isFusionInput()) {
    return;
  }
```

IIUC, we are only re-ordering this one if `it's a fusion input`. And my impression here is also that, in protocol it's also called on reference tensor and then propagated out. I was expecting that'll give us a consistent ordering for its consumers as well, even though this iterdomain doesn't map to anything afterwards.

Reply:
Ah right. Of course, since it's the reference tensor it will be propagated out so you don't need to worry about this. Thanks, this looks good to me.